pkg/controller/azvolume.go

```diff
@@ -423,7 +428,7 @@ func (r *ReconcileAzVolume) triggerUpdate(ctx context.Context, azVolume *azdiskv
 		}
 
 		//nolint:contextcheck // final status update of the CRI must occur even when the current context's deadline passes.
-		_ = azureutils.UpdateCRIWithRetry(context.Background(), nil, r.controllerSharedState.cachedClient, r.controllerSharedState.azClient, azVolume, updateFunc, consts.ForcedUpdateMaxNetRetry, azureutils.UpdateCRIStatus)
+		_, _ = azureutils.UpdateCRIWithRetry(goCtx, nil, r.controllerSharedState.cachedClient, r.controllerSharedState.azClient, azVolume, updateFunc, consts.ForcedUpdateMaxNetRetry, azureutils.UpdateCRIStatus)
```

Comment:
`The `goCtx` here is currently tied to the context passed to `triggerUpdate`. Does it need to be? If yes, then this must be a background context to ensure the final CRI state is updated if the deadline is exceeded.

Reply:
I thought I made it so that all `goCtx` are made from `context.Background()` but I just realized that one in `triggerExpand` was an exception. I will update this part.